src/filesystem/impls/filer/FilerFileSystem.js

```diff
@@ -180,34 +180,25 @@ define(function (require, exports, module) {
         oldPath = decodePath(oldPath);
         newPath = decodePath(newPath);
 
-        function updateBlobURL(err) {
+        function updateURL(err) {
             if(err) {
                 return callback(_mapError(err));
             }
 
-            // If this was a rename on a file path, update the Blob cache too
-            stat(newPath, function(err, stat) {
+            UrlCache.rename(oldPath, newPath, function(err, removed) {
                 if(err) {
                     return callback(_mapError(err));
                 }
 
-                if(stat.isFile) {
-                    UrlCache.rename(oldPath, newPath, function(err) {
-                        if(err) {
-                            return callback(_mapError(err));
-                        }
+                removed.forEach(function(pathInfo) {
+                    BrambleEvents.triggerFileRenamed(pathInfo.oldPath, pathInfo.newPath);
```

Comment:
Looks like we handle the event part at the API level: https://github.com/mozilla/brackets/blob/master/src/bramble/client/main.js#L711-L751. We stat the directory, and if it is a folder, we handle the event stuff there in the API (we only send one event so that we don't have to make multiple calls to publish for a simple rename), if it is a file, we offload the event handling to this piece of code. So this change will probably break that. One thing that you can do is run the `stat` call here before we do the rename and then trigger the appropriate event.